### ISSUE TITLE
ReplicationRule.getFilter() returning null for CRR

### ISSUE THREAD

**Issue by pikusayantan** [aws/aws-sdk-java#2433]
## Describe the bug

I am trying to retrieve Replication configuration for my S3 bucket using aws-java-sdk-s3 library.
But, when I am trying to fetch ReplicationFilter, the object is null.
My S3 bucket is having prefix added in the replication configuration.

## Expected Behavior

ReplicationFilter to return the ReplicationFilterPredicate. The predicate is one of ReplicationPrefixPredicate, ReplicationTagPredicate or ReplicationAndOperator type. 

## Current Behavior

Currently I am not able to retrieve Prefix or Tag used in our S3 bucket replication(CRR) configuration.
But deprecated method(getPrefix) in com.amazonaws.services.s3.model.ReplicationRule.getPrefix() returning proper value. 

## Steps to Reproduce

Create two S3 bucket in any two different region 1 and region 2 (eg, eu-west-1 and eu-west-2)
Add replication rule --> Set Source --> Prefix or Tag -> add any prefix
Save replication rule after adding other necessary details.

now using Java SDK 
```
AmazonS3 s3Client = ...
BucketReplicationConfiguration = BucketReplicationConfiguration = s3Client.getBucketReplicationConfiguration(bucketName);
for(Entry<String, ReplicationRule> entry : bucketReplicationConfiguration.getRules().entrySet()) {
  System.out.println(entry.getValue().getFilter()); //this is null
  System.out.println(entry.getValue().getFilter().getPredicate()); //this will throw nullpointer, this is where I like to fetch prefix/tag configured
}
```

## Your Environment

* AWS Java SDK version used: 1.11.874
* JDK version used: 1.8
* Operating System and version: MacOS Catalina

**Comment by debora-ito** [comment]
@pikusayantan I was not able to reproduce the issue, I can successfully retrieve the predicate content using your code sample.

Could you provide the wire logs of the operation call? Hopefully we can get additional information about what is being sent in the request and response. Make sure you redact all the sensitive information (like access key, bucket names).
https://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/java-dg-logging.html#sdk-net-logging-verbose

**Comment by pikusayantan** [comment]
Hi @debora-ito , I would not be able to share full logs due to some restrictions.
Meanwhile I tried setting log level for org.apache.http.wire to debug.
In log I cloud see response
```
<ReplicationConfiguration>
  <Rule>
    <ID>myID</ID>
    <Status>Enabled</Status>
    <Prefix>myPrefix/<Prefix>
    <Destination>...</Destination>
  </Rule>
</ReplicationConfiguration>
```

Another thing what I noticed in setFilter method there is check if prefix != null. This would be the cause `ReplicationFilter` being null. This made me to use deprecated method getPrefix() to retrieve my replication prefix configuration.
https://github.com/aws/aws-sdk-java/blob/5694fc578e5069dbadb40b789091d4db9a834edc/aws-java-sdk-s3/src/main/java/com/amazonaws/services/s3/model/ReplicationRule.java#L215-L225

Not sure that is the intended behaviour. 
When I was working with S3 Lifecycle. To retrieve s3 lifecycle prefix/tag I used `BucketLifecycleConfiguration` in which LifecycleFilter is not null. Both has similar pattern to retrieve prefix and tags apart from setter has null checks in replication
https://github.com/aws/aws-sdk-java/blob/5694fc578e5069dbadb40b789091d4db9a834edc/aws-java-sdk-s3/src/main/java/com/amazonaws/services/s3/model/BucketLifecycleConfiguration.java#L550-L552
